### PR TITLE
Support Mantic compilers

### PR DESCRIPTION
**Pull request by luis4a0**
This fixes compilation with gcc-13 and clang-16.

### CODE REVIEW

**Comment by luis4a0** [comment]
Still having issues with the `fmt` lib with gcc-13.

**Comment by luis4a0** [comment]
Done.

**Comment by luis4a0** [comment]
Tested it also works with clang-17.

File: 3rd-party/fmt
Comment (townsend2010) [inline 3rd-party/fmt:1]: It looks like this is an update to the `fmt` submodule.  Please update the `3rd-party/submodule_info.md` file to reflect this?

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Seems fine overall, but I have one inline comment and a general question.  Why the need to add those escaped quotes in some of the strings?

**Comment by luis4a0** [comment]
Hey @townsend2010, thanks! I added those escape chars because, for some change in `fmt`, some strings which were outputted with double quotes don't have double quotes now. I wanted the output to be the same as before the change in `fmt`.

File: 3rd-party/fmt
Comment (luis4a0) [reply 3rd-party/fmt:1]: Done. Switched to version 10.1.1 (the last stable) and reflected it in `submodule_info.md`.

**Comment by luis4a0** [comment]
Hey @townsend2010! The default formatting of `std::filesystem::path` changed in the last version of `fmtlib`. It used to quote the paths, and now it defaults to not quoting them. Thus I forced quoting where needed, so the tests are not affected by the change.

**Review by townsend2010 — APPROVED** [review]
Ok, LGTM, thanks!

**Pull request merged**